docs/blog/_posts/2020-12-18-scala3-m3.md

```diff
@@ -0,0 +1,209 @@
+Previously, context function closures were written as follows:
+
+```scala
+(using s: Show[String]) => s.show("foobar")
+```
+
+Now, we have reverted to the following syntax:
+```scala
+(s: Show[String]) ?=> s.show("foobar")
+```
+
+# `Matchable` trait – a new top-level type
+The motivation for this change is the fact that it is possible to access the underlying type referenced by an opaque type using pattern matching:
+
+```scala
+val imm = IArray(1,2,3) // supposedly immutable...
+imm match
+   case a: Array[Int] => a(0) = 0  // but that's shown to be lie
+```
+
+To address this change, a new trait, `Matchable`, is introduced at the top of the type hierarchy:
+
+```scala
+abstract class Any:
+  def asInstanceOf
+  def ==
+  def !=
+  def ##
+  def equals
+  def hashCode
+  def toString
+
+trait Matchable extends Any:
+  def isInstanceOf
+  def getClass
+
+class AnyVal extends Any, Matchable
+
+class Object extends Any, Matchable
+```
+
+Matchable is currently a marker trait without any methods. Over time we might migrate methods getClass and isInstanceOf to it, since these are closely related to pattern-matching.
+
+With the addition of this type, we now make the compiler emit warnings when trying to do a trick like the one discussed above:
+
+```scala
+-- Warning: i7314.scala:6:12 ---------------------------------------------------
+6 |    case a: Array[Int] =>
+  |            ^^^^^^^^^^
+  |            pattern selector should be an instance of Matchable,
+  |            but it has unmatchable type opaques.IArray[Int] instead
+```
+
+The change involved an extensive discussion which you can read in the [PR #10670](https://github.com/lampepfl/dotty/pull/10670). Read more about this change in the [documentation](https://dotty.epfl.ch/docs/reference/other-new-features/matchable.html).
+
+# Tooling improvements
+As the work on Scala 3 gets closer to the stable release, the focus increasingly shifts on the tooling available to get started with Scala 3.
+
+For a while now, we are not using the old dottydoc documentation tool for building the documentation. We are developing an entirely new tool, scala3doc, from scratch.
+
+As part of the tooling effort, this new Scala 3 documentation tool is rapidly improved. [PR #10522](https://github.com/lampepfl/dotty/pull/10522) proves that the doctool can generate documentation for the community build projects.
+
+[PR #10491](https://github.com/lampepfl/dotty/pull/10491) introduced the scripting capability to Scala 3. Consider the following source named e.g. `Main.scala`:
+
+```scala
+@main def Test(name: String): Unit =
+  println(s"Hello ${name}!")
+```
+
+If you have Scala 3 binaries on your path (which you can get by following the steps at the [Dotty website](https://dotty.epfl.ch/)'s section "Try Dotty"), you can run the `scala Main.scala World` command. This command will compile the source in question to a temporary directory and run the discovered main method with the argument `World`.
```

Comment:
```suggestion
If you have Scala 3 binaries on your path (which you can get by following the steps on the [Dotty website](https://dotty.epfl.ch/), in the section "Try Dotty"), you can run the following command:

    $ scala Main.scala World

This will compile the source in question to a temporary directory and run the discovered main method with the argument `World`.
```